Include/objimpl.h

```diff
@@ -138,6 +138,9 @@ _PyObject_INIT(PyObject *op, PyTypeObject *typeobj)
 {
     assert(op != NULL);
     Py_TYPE(op) = typeobj;
+    if (PyType_GetFlags(typeobj) & Py_TPFLAGS_HEAPTYPE) {
+        Py_INCREF(typeobj);
```

Comment:
I don't understand why "convert statically allocated types (DirEntryType & ScandirIteratorType) to heap-allocated  types" needs to modify the very important _PyObject_INIT() function. If you want to modify it, would it be possible to do that in a first PR which only modify _PyObject_INIT() (with related changes)?

Reply:
Yeah, that makes sense! I just created: 
https://bugs.python.org/issue35810
https://github.com/python/cpython/pull/11661

I will rebase this change once that PR is merged.